### PR TITLE
test: unit test the intacct export log pages

### PR DESCRIPTION
**Pull request by JustARatherRidiculouslyLongUsername**
image

## Clickup
https://app.clickup.com/t/86cwh86cx
https://app.clickup.com/t/86cwh86cu

## Release Notes

- **New Features**
	- Enhanced flexibility in method calls with optional parameters for date filters and search queries.
	- Improved handling of pagination and date filters in the export log components.

- **Bug Fixes**
	- Adjusted pagination logic to ensure accurate page calculations based on the current limit.

- **Tests**
	- Expanded test coverage for `IntacctCompletedExportLogComponent` and `IntacctSkipExportLogComponent`, including various functionalities like initialization, pagination, and search handling.

- **Chores**
	- Updated mock data structures to improve testing for expense groups and export logs.

### CODE REVIEW

File: src/app/integrations/intacct/intacct-main/intacct-export-log/intacct-completed-export-log/intacct-completed-export-log.component.spec.ts [hunk 1]
```diff
@@ -1,23 +1,143 @@
...
+    paginatorService = TestBed.inject(PaginatorService) as jasmine.SpyObj<PaginatorService>;
+    userService = TestBed.inject(UserService) as jasmine.SpyObj<UserService>;
+
+    userService.getUserProfile.and.returnValue({ org_id: 'ORG123' } as MinimalUser);
+    paginatorService.getPageSize.and.returnValue(mockPaginator);
+    exportLogService.getExpenseGroups.and.returnValue(of(mockExpenseGroupResponse));
 
     fixture = TestBed.createComponent(IntacctCompletedExportLogComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
```
Comment (DhaaraniCIT) [inline src/app/integrations/intacct/intacct-main/intacct-export-log/intacct-completed-export-log/intacct-completed-export-log.component.spec.ts:17]: why is this line removed?

File: src/app/integrations/intacct/intacct-main/intacct-export-log/intacct-completed-export-log/intacct-completed-export-log.component.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ export class IntacctCompletedExportLogComponent implements OnInit {
 
   dateOptions: DateFilter[] = AccountingExportModel.getDateOptionsV2();
 
-  selectedDateFilter: SelectedDateFilter | null;
+  selectedDateFilter?: SelectedDateFilter | null;
```
Comment (DhaaraniCIT) [inline src/app/integrations/intacct/intacct-main/intacct-export-log/intacct-completed-export-log/intacct-completed-export-log.component.ts:45]: why are we add '?' here

File: src/app/integrations/intacct/intacct-main/intacct-export-log/intacct-skip-export-log/intacct-skip-export-log.component.ts [hunk 3]
```diff
@@ -40,7 +40,7 @@ export class IntacctSkipExportLogComponent implements OnInit {
 
   dateOptions: DateFilter[] = AccountingExportModel.getDateOptionsV2();
 
-  selectedDateFilter: SelectedDateFilter | null;
+  selectedDateFilter?: SelectedDateFilter | null;
```
Comment (DhaaraniCIT) [inline src/app/integrations/intacct/intacct-main/intacct-export-log/intacct-skip-export-log/intacct-skip-export-log.component.ts:43]: same as above

File: src/app/integrations/intacct/intacct-main/intacct-export-log/intacct-skip-export-log/intacct-skip-export-log.component.ts [hunk 4]
```diff
@@ -60,7 +60,7 @@ export class IntacctSkipExportLogComponent implements OnInit {
 
   readonly brandingConfig = brandingConfig;
 
-  searchQuery: string | null;
+  searchQuery?: string | null;
```
Comment (DhaaraniCIT) [inline src/app/integrations/intacct/intacct-main/intacct-export-log/intacct-skip-export-log/intacct-skip-export-log.component.ts:63]: same as above

File: src/app/integrations/intacct/intacct-main/intacct-export-log/intacct-skip-export-log/intacct-skip-export-log.component.ts [hunk 5]
```diff
@@ -89,10 +89,10 @@ export class IntacctSkipExportLogComponent implements OnInit {
 
   pageSizeChanges(limit: number): void {
     this.isLoading = true;
-    this.limit = limit;
     this.currentPage = 1;
     this.selectedDateFilter = this.selectedDateFilter ? this.selectedDateFilter : null;
     this.getSkippedExpenses(limit, this.offset);
+    this.limit = limit;
```
Comment (DhaaraniCIT) [inline src/app/integrations/intacct/intacct-main/intacct-export-log/intacct-skip-export-log/intacct-skip-export-log.component.ts:95]: why we change the position of the limit, any reason

Comment (JustARatherRidiculouslyLongUsername) [reply src/app/integrations/intacct/intacct-main/intacct-export-log/intacct-skip-export-log/intacct-skip-export-log.component.ts:95]: Yes, since calling `this.getSkippedExpenses` with `limit` and `this.limit` set to the same value prevents it from updating the localstorage with the new value because of [this check](https://github.com/fylein/fyle-integrations-app/blob/master/src/app/integrations/intacct/intacct-main/intacct-export-log/intacct-completed-export-log/intacct-completed-export-log.component.ts#L118). Essentially that leaves the localstorage functionality unused in this page. Calling `this.getSkippedExpenses` *before* updating `this.limit` updates the localStorage value as intended.

I've also raised a separate PR (#990) for this

**Review by DhaaraniCIT — APPROVED** [review]

**Review by ashwin1111 — APPROVED** [review]
